lib/protocol/message.go

```diff
@@ -148,6 +156,13 @@ type Device struct {
 	Options         []Option // max:64
 }
 
+type FileDownloadProgressUpdate struct {
```

Comment:
The current BEP spec, mostly the Request message, appears to me to rely on folder name, file name, block offset, and block hash for identifying a block. What drove the desire to refer to blocks here with folder name, file name, file version, and block index number? It seems like a disconnect for me.

I'm not necessarily opposed to it but am currently curious. I also haven't been able to review this thoroughly enough for it to be immediately obvious right now.